plugins/woocommerce/src/Internal/Abilities/REST/RestAbilityFactory.php

```
@@ -0,0 +1,476 @@
+	}
+
+	/**
+	 * Execute the REST operation.
+	 *
+	 * @param object $controller REST controller instance.
+	 * @param string $operation Operation type.
+	 * @param array  $input Input parameters.
+	 * @param string $route REST route template with named placeholders.
+	 * @return mixed Operation result.
+	 */
+	private static function execute_operation( $controller, string $operation, array $input, string $route ) {
```

PHPMD (2.15.0)

Avoid unused parameters such as '$controller'. (undefined)

(UnusedFormalParameter)